core/app/models/spree/promotion/rules/nth_order.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  class Promotion
+    module Rules
+      class NthOrder < PromotionRule
+        preference :nth_order, :integer, default: 2
+        # It does not make sense to have this apply to the first order using preferred_nth_order == 1
+        # Instead we could use the first_order rule
+        validates :preferred_nth_order, numericality: { only_integer: true, greater_than: 1 }
+
+        # This promotion is applicable to orders only.
+        def applicable?(promotable)
+          promotable.is_a?(Spree::Order)
+        end
+
+        # This is never eligible if the order does not have a user, and that user does not have any previous completed orders.
+        #
+        # Use the first order rule if you want a promotion to be applied to the first order for a user.
+        # @param order [Spree::Order]
+        # @option options
+        def eligible?(order, options = {})
+          return false unless order.user && has_completed_orders?(order.user)
+          nth_order?(order.user)
+        end
+
+        private
+
+        def completed_order_count user
+          user.orders.complete.count
```

Comment:
Thinking this should maybe be the count of orders completed before this order. That way this could be realculated even on a completed order correctly. This will probably work without that because the adjustment will be closed on order completion, but we shouldn't depend on that.